src/modules/hip/kernel/noise_gaussian.hpp

```diff
@@ -253,6 +273,102 @@ __global__ void gaussian_noise_pln3_pkd3_hip_tensor(T *srcPtr,
+    if(!copyInput)
+    {
+        for(int c = 0; c < channels; c++)
+        {
+            rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &val_f8);
+            gaussian_noise_3d_8_hip_compute(&val_f8, &xorwowState, gaussianNoise3dParams_f2.x, gaussianNoise3dParams_f2.y);
+            rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &val_f8);
+            srcIdx += srcStridesCDH.x;
+            dstIdx += dstStridesCDH.x;
+        }
+    }
+    else
+    {
+        for(int c = 0; c < channels; c++)
+        {
+            rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &val_f8);
+            rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &val_f8);
+            srcIdx += srcStridesCDH.x;
+            dstIdx += dstStridesCDH.x;
+        }
+    }
+}
+
+__global__ void gaussian_noise_3d_ndhwc_hip_tensor(float *srcPtr,
+                                                   uint2 srcStridesDH,
+                                                   float *dstPtr,
+                                                   uint2 dstStridesDH,
+                                                   float2 gaussianNoise3dParams_f2,
+                                                   RpptXorwowStateBoxMuller *xorwowInitialStatePtr,
+                                                   uint *xorwowSeedStream,
+                                                   RpptROI3DPtr roiGenericSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // WC - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= roiGenericSrc->xyzwhdROI.roiDepth) || (id_y >= roiGenericSrc->xyzwhdROI.roiHeight) || (id_x >= roiGenericSrc->xyzwhdROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = ((id_z + roiGenericSrc->xyzwhdROI.xyz.z) * srcStridesDH.x) + ((id_y + roiGenericSrc->xyzwhdROI.xyz.y) * srcStridesDH.y) + (id_x + roiGenericSrc->xyzwhdROI.xyz.x) * 3;
+    uint dstIdx = (id_z * dstStridesDH.x) + (id_y * dstStridesDH.y) + id_x * 3;
+    uint seedStreamIdx = (id_y * dstStridesDH.y) + (hipBlockIdx_x * hipBlockDim_x) + hipThreadIdx_x;
+
+    RpptXorwowStateBoxMuller xorwowState;
+    uint xorwowSeed = xorwowSeedStream[seedStreamIdx % SEED_STREAM_MAX_SIZE];
+    xorwowState.x[0] = xorwowInitialStatePtr->x[0] + xorwowSeed;
+    xorwowState.x[1] = xorwowInitialStatePtr->x[1] + xorwowSeed;
+    xorwowState.x[2] = xorwowInitialStatePtr->x[2] + xorwowSeed;
+    xorwowState.x[3] = xorwowInitialStatePtr->x[3] + xorwowSeed;
+    xorwowState.x[4] = xorwowInitialStatePtr->x[4] + xorwowSeed;
+    xorwowState.counter = xorwowInitialStatePtr->counter + xorwowSeed;
+
+    d_float24 val_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &val_f24);
+    bool copyInput = (!gaussianNoise3dParams_f2.x) && (!gaussianNoise3dParams_f2.y);
```

Comment:
When does gaussian noise give output = input? Isn't it for mean = 0 and stdDev = 1? Pls check condition